### PR TITLE
Try to fix node 12 compilation issues

### PR DESCRIPTION
**Pull request by lukasz-jagielski-bolt**
# Issue

Hi, this PR fixes compilation issues against Node v12. Mostly mechanical changes to use nan wrappers where internal deprecated v8 APIs were used. I don't have much time to push this through so if someone decides to take over, I won't be offended :)
Thanks,
Łukasz 

## Tasklist

 - [ ] CHANGELOG.md entry ([How to write a changelog entry](http://keepachangelog.com/en/1.0.0/#how))
 - [ ] update relevant [Wiki pages](https://github.com/Project-OSRM/osrm-backend/wiki)
 - [ ] add tests (see [testing documentation](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/testing.md)
 - [ ] review
 - [ ] adjust for comments
 - [ ] cherry pick to release branch

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

**Comment by mihneadb** [comment]
I see the build failed with a HTTP 403. Can it be retriggered or is it something else?

**Comment by mihneadb** [comment]
Hi, any input on this? Thanks!

**Comment by mihneadb** [comment]
@danpat maybe you can take a look at this? Thanks!

**Comment by mihneadb** [comment]
Hi, any updates? Thanks!

File: include/nodejs/node_osrm_support.hpp [hunk 1]
```diff
@@ -1095,9 +1094,9 @@ argumentsToTileParameters(const Nan::FunctionCallbackInfo<v8::Value> &args, bool
         return tile_parameters_ptr();
     }
```
Comment (akashihi) [inline include/nodejs/node_osrm_support.hpp:1096]: I'm not a Node guru, so just curious: is behavior changed here? IIUC calling FromJust() with missing values will just blow the app. Will ` x->Uint32Value();` act same?

**Comment by mihneadb** [comment]
`make tests` fails at the point where it's trying to download `https://s3.amazonaws.com/mapbox/osrm/testing/monaco.poly`. However I do get the following test binaries:
```
engine-tests
library-contract-tests
library-customize-tests
library-partition-tests
server-tests
```
I ran all of them and they ran successfully, yes.

File: include/nodejs/node_osrm_support.hpp (hunk 1 above)
Comment (lukasz-jagielski-bolt) [reply include/nodejs/node_osrm_support.hpp:1096]: Hi, thanks for taking a look! So Above, there's a guard which checks that x,y,z hold uint32_t value, so calling it should never cause a problem

**Comment by akashihi** [comment]
tests are fixed with #5729 , which is not yet merged :( I pinged reviewers on that other ticket, let's wait couple of days for it.

**Comment by akashihi** [comment]
As CI is broken, i ran all the tests locally and everything seems to be fine:
1283 scenarios (1283 passed)
6826 steps (6826 passed)

**Pull request merged**